streaming/src/main/scala/io/gearpump/streaming/appmaster/TaskManager.scala

```diff
@@ -44,12 +44,12 @@ import scala.concurrent.duration._
 private[appmaster] class TaskManager(
     appId: Int,
     dagManager: ActorRef,
-    subDAGManager: SubDAGManager,
+    jarScheduler: JarScheduler,
     executorManager: ActorRef,
     clockService: ActorRef,
     appMaster: ActorRef,
     appName: String)
-  extends Actor with FSM[State, StateData] {
+  extends Actor {
```

Comment:
Several reasons:
1. Equivalent in function, as least this is true for TaskManager.
2. We can have more than one state, like this: dynamicDag(state, recoverState), so that no need to hack the StateData to contain recoverState. 
3. Receive is more easy and composable than StateFunction. 
For example
```
case Event(StartExecutorsTimeOut, state)
```
it is not necessary for every event like StartExecutorsTimeOut to coupled with a state. If I want to reuse this code, then I need to make sure the state points to same type.

4. There is no type in State.
case Event(StartExecutorsTimeOut, state)
I need to cast state to some sub-class to use it. I don't like casting.